zqd/search/indexsearch.go

```diff
@@ -6,26 +6,22 @@ import (
 	"github.com/brimsec/zq/archive"
 	"github.com/brimsec/zq/zbuf"
 	"github.com/brimsec/zq/zqd/api"
-	"github.com/brimsec/zq/zqd/space"
-	"github.com/brimsec/zq/zqd/storage/archivestore"
-	"github.com/brimsec/zq/zqe"
 )
 
 type IndexSearch struct {
 	zbuf.ReadCloser
 }
 
-func NewIndexSearch(ctx context.Context, s *space.Space, req api.IndexSearchRequest) (*IndexSearch, error) {
-	arkstore, ok := s.Storage.(*archivestore.Storage)
-	if !ok {
-		return nil, zqe.E(zqe.Invalid, "index search only supported on archive spaces")
-	}
+type IndexedArchive interface {
+	IndexSearch(context.Context, archive.IndexQuery) (zbuf.ReadCloser, error)
+}
```

Comment:
Nits: I think IndexSearcher would be a more descriptive name for this interface. Also, declare it before IndexSearch so IndexSearch remains adjacent to its constructor, NewIndexSearch.

Reply:
as discussed 1x1, going to use IndexSearcher for the interface name & IndexSearchOp for the struct.